runtime/src/main/cpp/Runtime.cpp

```diff
@@ -76,7 +76,44 @@ inline bool isValidRuntime() {
 
 volatile int aliveRuntimesCount = 0;
 
+enum GlobalRuntimeStatus {
+    kGlobalRuntimeUninitialized = 0,
+    kGlobalRuntimeRunning,
+    kGlobalRuntimeShuttingDown,
+    kGlobalRuntimeShutdown,
+};
+
+volatile GlobalRuntimeStatus globalRuntimeStatus = kGlobalRuntimeUninitialized;
+
+class ScopedInitializingRuntime {
+public:
+    ScopedInitializingRuntime() { atomicAdd(&initializingRuntimesCount, 1); }
+
+    ~ScopedInitializingRuntime() { atomicAdd(&initializingRuntimesCount, -1); }
+
+    static bool IsInitializing() { return atomicGet(&initializingRuntimesCount) > 0; }
+
+    ScopedInitializingRuntime(const ScopedInitializingRuntime&) = delete;
+    ScopedInitializingRuntime(ScopedInitializingRuntime&&) = delete;
+    ScopedInitializingRuntime& operator=(const ScopedInitializingRuntime&) = delete;
+    ScopedInitializingRuntime& operator=(ScopedInitializingRuntime&&) = delete;
```

Comment:
Is there a C++ magic spell that would make this automatically? :)
Maybe extracting a `Scope*` superclass or something.

Reply:
Probably. It's also enough to delete just copy-ctor + copy-assignment, since that automatically deletes move-ctor + move-assignment. However, I don't see anything wrong with being explicit here.

Reply:
> However, I don't see anything wrong with being explicit here.

Too explicit. This is basically a boilerplate.

Reply:
Removed the class completely.